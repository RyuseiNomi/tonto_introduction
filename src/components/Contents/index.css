.content {
  background-color: #E89A8A;
}

.content p {
  color: white;
}

.content .content-title {
  display: flex;
  align-items: center;
  justify-content: center;
}

.content .link-button {
  background-color: white;
  color: #E89A8A;
  padding: 10px 20px 10px 20px;
  border-radius: 20px;
  font-weight: bold;
}

/* ========= スマホサイズ_1 =========== */
@media (min-width: 280px) {
  .content .content-title {
    padding: 20px 0 20px 0;
  }

  .content .content-title p {
    border-bottom: solid 6px white;
    max-width: 40%;
    font-size: 20pt;
  }

  .content-image {
    padding: 30px 0 10px 0;
  }

  .content-image img {
    max-width: 180px;
    max-height: 180px
  }

  .content-description {
    padding: 0 20px 20px 20px;
  }

  .content-description .content-description-title {
    padding: 10px 0 10px 0;
    font-size: 18pt;
  }

  .content-description .content-description-detail {
    padding: 10px 0 10px 0;
    font-size: 11pt;
  }

  .link {
    padding: 20px 0 20px 0;
  }
}

/* ========= スマホサイズ_2 =========== */
@media (min-width: 576px) {
  .content .content-title {
    padding: 40px 0 30px 0;
  }

  .content .content-title p {
    border-bottom: solid 6px white;
    max-width: 50%;
    font-size: 36pt;
  }

  .content-image {
    padding: 30px 0 10px 0;
  }

  .content-image img {
    max-width: 240px;
    max-height: 240px
  }

  .content-description {
    padding: 0 80px 50px 80px;
  }

  .content-description .content-description-title {
    padding: 10px 0 10px 0;
    font-size: 24pt;
  }

  .content-description .content-description-detail {
    padding: 10px 0 10px 0;
    font-size: 12pt;
  }

  .link {
    padding: 20px 0 20px 0;
  }
}

/* ========= タブレットサイズ =========== */
@media (min-width: 768px) {
  .content .content-title {
    padding: 50px 0 40px 0;
  }

  .content .content-title p {
    border-bottom: solid 10px white;
    max-width: 50%;
    font-size: 40pt;
  }

  .content-image {
    display: flex;
    padding: 50px 10px 50px 0;
    justify-content: right;
  }

  .content-image img {
    max-width: 200px;
    max-height: 200px
  }

  .content-description {
    text-align: left;
    padding: 50px 30px 50px 10px;
  }

  .content-description .content-description-title {
    padding: 0 0 10px 0;
    font-size: 24pt;
  }

  .content-description .content-description-detail {
    padding: 10px 0 10px 0;
    font-size: 12pt;
  }
}

/* ========= PCサイズ =========== */
@media (min-width: 992px) {
  .content .content-title {
    padding: 50px 0 40px 0;
  }

  .content .content-title p {
    border-bottom: solid 10px white;
    max-width: 50%;
    font-size: 40pt;
  }

  .content-image {
    display: flex;
    padding: 50px 20px 50px 0;
    justify-content: right;
  }

  .content-image img {
    max-width: 240px;
    max-height: 240px
  }

  .content-description {
    text-align: left;
    padding: 50px 100px 50px 10px;
  }

  .content-description .content-description-title {
    padding: 0 0 10px 0;
    font-size: 30pt;
  }

  .content-description .content-description-detail {
    padding: 10px 0 10px 0;
    font-size: 14pt;
  }
}

/* ========= PCサイズ_2 =========== */
@media (min-width: 1120px) {
  .content-image {
    display: flex;
    padding: 50px 30px 50px 0;
    justify-content: right;
  }

  .content-image img {
    max-width: 260px;
    max-height: 260px
  }
}
