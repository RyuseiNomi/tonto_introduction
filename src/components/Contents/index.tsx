import React from 'react';
import './index.css';
import { Row, Col } from 'antd';

type Content = {

  /* 成果物のイメージ画像( webp とそれ以外の拡張子画像) */
  image: {
    webp: string;
    others: string;
  },
  /* 成果物のタイトル */
  title: {
    jp: string;
    en: string;
  }
  /* 成果物の説明文 */
  description: {
    jp: string;
    en: string;
  },
  /* 成果物のリンク */
  url: string;
}

type Props = {
  contents: Content[]
  language: string;
}

function Contents(props: Props) {

  const { contents, language } = props;

  return (
    <div className="content">
      <div className="content-title"><p>CONTENTS</p></div>
      {contents.map((content, index) => {
        const { image, title, description, url } = content;
        return (
          <Row>
            <Col xs={24} sm={24} md={8} lg={10} >
              <div className="content-image">
                <picture>
                  <source type="image/webp" srcSet={`${process.env.PUBLIC_URL + image.webp}`} ></source>
                  <img src={`${process.env.PUBLIC_URL + image.others}`} alt="thumbnail" />
                </picture>
              </div>
            </Col>
            <Col xs={24} sm={24} md={16} lg={14} >
              <div className="content-description">
                <p className="content-description-title">{language === "jp" ? title.jp : title.en}</p>
                <p className="content-description-detail">{language === "jp" ? description.jp : description.en}</p>
                <a href={url} rel="noopener noreferrer" target="_blank">{url}</a>
              </div>
            </Col>
          </Row>
        );
      })}
    </div>
  );
}

export default Contents;
