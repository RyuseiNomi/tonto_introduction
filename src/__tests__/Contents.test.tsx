import { act } from "react-dom/test-utils";
import { render, unmountComponentAtNode } from "react-dom";
import renderer from "react-test-renderer";
import Contents from "./../components/Contents";

let container = null; // ダミーの DOM を格納するためのコンテナ
const sample_content = [{
  image: {webp: "sample_webp_image.webp", others: "sample_png_image.png"},
  title: {jp: "サンプル日本語タイトル", en: "sample english title"},
  description: {jp: "サンプル日本語説明文", en: "sample english description"},
  url: "http://hoge.fuga"
}];

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // 作成したダミーの DOM を除去する
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

// =============== snapshot test =================
it('コンテンツ群が表示されていること', () => {
  const component = renderer.create(<Contents contents={sample_content} />);
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

// =============== tests with act =================
it('Row と Col が表示されていること', () => {
  act(() => {
    render(<Contents contents={sample_content} />, container);
  });

  // NOTE: Row と Col がモック化されている以上、同じレイアウトを検証することは難しい。
  // そのため、Unit Test ではモックが呼ばれていることを担保することに留めている
  // TODO: antd を利用しないよう修正をしたため、テストも細かく書く
  //const row = document.querySelector("[data-testid=row]");
  //expect(row.textContent).toBe('Row');
});
