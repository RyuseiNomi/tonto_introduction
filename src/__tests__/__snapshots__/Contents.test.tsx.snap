// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`コンテンツ群が表示されていること 1`] = `
<div
  className="contents"
>
  <p
    className="content-title"
  >
    CONTENTS
  </p>
  <div
    className="content"
  >
    <picture>
      <source
        srcSet="sample_webp_image.webp"
        type="image/webp"
      />
      <img
        alt="thumbnail"
        src="sample_png_image.png"
      />
    </picture>
    <div
      className="content-description"
    >
      <p
        className="content-description-title"
      >
        sample english title
      </p>
      <p
        className="content-description-detail"
      >
        sample english description
      </p>
      <div
        className="link"
      >
        <a
          className="link-button"
          href="http://hoge.fuga"
          rel="noopener noreferrer"
          target="_blank"
        >
          Go website
        </a>
      </div>
    </div>
  </div>
</div>
`;
